### PR TITLE
implement ABitOfEverythingService

### PR DESCRIPTION
**Pull request by mattn**

### CODE REVIEW

File: examples/server/main.go [hunk 1]
```diff
@@ -1,23 +1,98 @@
...
+		m: make(map[string]*examples.ABitOfEverything),
+	}
+}
+
+func (s *_ABitOfEverythingServer) Create(ctx context.Context, msg *examples.ABitOfEverything) (*examples.ABitOfEverything, error) {
+	glog.Info(msg)
+	var uuid string
+	for {
+		uuid = fmt.Sprintf("%x", uuidgen.Next())
+		if _, ok := s.m[uuid]; !ok {
```
Comment (yugui) [inline examples/server/main.go]: race condition?

File: examples/server/main.go [hunk 2]
```diff
@@ -1,23 +1,98 @@
...
+	return s.m[uuid], nil
+}
+
+func (s *_ABitOfEverythingServer) CreateBody(ctx context.Context, msg *examples.ABitOfEverything) (*examples.ABitOfEverything, error) {
+	return s.Create(ctx, msg)
+}
+
+func (s *_ABitOfEverythingServer) Lookup(ctx context.Context, msg *examples.IdMessage) (*examples.ABitOfEverything, error) {
+	glog.Info(msg)
+	if a, ok := s.m[msg.Uuid]; ok {
```
Comment (yugui) [inline examples/server/main.go]: data race?

File: examples/server/main.go [hunk 3]
```diff
@@ -1,23 +1,98 @@
...
+		return a, nil
+	}
+	return nil, errors.New("not found")
```
Comment (yugui) [inline examples/server/main.go]: use grpc.Errorf(codes.NotFound, "not found").

File: examples/server/main.go [hunk 4]
```diff
@@ -1,23 +1,98 @@
...
+}
+
+func (s *_ABitOfEverythingServer) Update(ctx context.Context, msg *examples.ABitOfEverything) (*examples.EmptyMessage, error) {
+	glog.Info(msg)
+	if _, ok := s.m[msg.Uuid]; ok {
+		s.m[msg.Uuid] = msg
+	} else {
+		return nil, errors.New("not found")
```
Comment (yugui) [inline examples/server/main.go]: ditto

File: examples/server/main.go [hunk 5]
```diff
@@ -1,23 +1,98 @@
...
+	}
+	return new(examples.EmptyMessage), nil
+}
+
+func (s *_ABitOfEverythingServer) Delete(ctx context.Context, msg *examples.IdMessage) (*examples.EmptyMessage, error) {
+	glog.Info(msg)
+	if _, ok := s.m[msg.Uuid]; ok {
+		delete(s.m, msg.Uuid)
+	} else {
+		return nil, errors.New("not found")
```
Comment (yugui) [inline examples/server/main.go]: ditto.

**Comment by yugui** [comment]
Thank you for your contribution.  Could you fix the minor issues before merge?

**Comment by mattn** [comment]
done

**Comment by mattn** [comment]
fixed race conditions

**Comment by yugui** [comment]
LGTM

**Comment by mattn** [comment]
need this? https://github.com/mattn/grpc-gateway/blob/62000f7e3bf9a8791ee05710e87b4d518d76f155/examples/server/main.go#L110

**Comment by hayate** [comment]
thank you @mattn LGTM

**Pull request merged**